### PR TITLE
test(epp): add unit tests for error package

### PR DESCRIPTION
**Pull request by nsega**
## What type of PR is this?
/kind cleanup

## What this PR does / why we need it:
This PR adds comprehensive unit tests for the `pkg/epp/util/error` package, which previously had no test coverage.

## Changes:
- Add comprehensive tests for Error.Error() method
- Add tests for CanonicalCode() function with edge cases
- Test all error code constants
- Cover 100% test coverage for pkg/epp/util/error

## Testing:
  - All new tests pass
  - Existing tests continue to pass
  - 100% test coverage achieved for the package

### CODE REVIEW

**Comment by kfswain** [comment]
Looks great! Thanks! Will approve once CI finishes (assuming it all passes)

**Comment by nsega** [comment]
@kfswain Thank you for your review!

I just addressed the CI Test failure, which was the Boilerplate header of the target test code was wrong [ref](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_gateway-api-inference-extension/1052/pull-gateway-api-inference-extension-verify-main/1937672370005217280)

Just in case, I verified locally using hack/verify-boilerplate.sh as well after adding the correct boilerplate header with the exact format used by the project.　

Since all CI passes successfully, can you please approve this at your convenience?

File: pkg/epp/util/error/error_test.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+	}
+
+	expectedValues := map[string]string{
+		"Unknown":                        "Unknown",
+		"BadRequest":                     "BadRequest",
+		"Internal":                       "Internal",
+		"ModelServerError":               "ModelServerError",
+		"BadConfiguration":               "BadConfiguration",
+		"InferencePoolResourceExhausted": "InferencePoolResourceExhausted",
+	}
```
Comment (danehans) [inline pkg/epp/util/error/error_test.go]: Why not use the consts as map keys:
```go
const (
	Unknown                        = "Unknown"
	BadRequest                     = "BadRequest"
	Internal                       = "Internal"
	ModelServerError               = "ModelServerError"
	BadConfiguration               = "BadConfiguration"
	InferencePoolResourceExhausted = "InferencePoolResourceExhausted"
)
```

Comment (nsega) [reply pkg/epp/util/error/error_test.go]: It makes sense to me. Since the constants and expected values I defined seem redundant, I updated the test case by simplifying it to define only one map as a test.

**Review by nsega — COMMENTED** [review]
@danehans Thank you for your feedback! I addressed it. Can you please take a look at it again?

File: pkg/epp/util/error/error_test.go (hunk 1 above)
Comment (nirrozenbaum) [reply pkg/epp/util/error/error_test.go]: these constants already defined in error.go. why not use reference them? why do we need to define them again?
https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/9c9abd51a6d0b27f7b4d6d11583e9d7bead2c756/pkg/epp/util/error/error.go#L29-L36

Comment (nsega) [reply pkg/epp/util/error/error_test.go]: @nirrozenbaum Based on  danehans's feedback the other day, I updated the test to check the error message using the constant. https://github.com/kubernetes-sigs/gateway-api-inference-extension/pull/1052/commits/35d2f231cb95c36f5a908652ea852d38908d2edc How does it sound to you?

Comment (nirrozenbaum) [reply pkg/epp/util/error/error_test.go]: I was commenting on the previous comment, sorry. 

I think overall the code looks great, just not sure why unit-tests for constants are needed :)
Constants are compile-time values, there's no logic or behavior to test here — it's static data.

Unit tests are meant to test behavior and logic, not to assert that literals haven't changed (unless those values are dynamically generated or critical public APIs).

Comment (nsega) [reply pkg/epp/util/error/error_test.go]: @nirrozenbaum No worries. I agree with your point. In general, we don't need to write unit tests for constants like you said. However, I do think that testing the literal values of exported constants is a valid and important test when those constants are part of the public API, potentially used by external systems, or part of error messages that might be parsed. This is the reason why I added the case.

Comment (nirrozenbaum) [reply pkg/epp/util/error/error_test.go]: that's also fine to keep them :)

**Comment by nirrozenbaum** [comment]
/lgtm
/approve

Congrats @nsega on your first inference gateway PR!

**Comment by nsega** [comment]
@nirrozenbaum Thank you very much for your kind review! I was excited to contribute to inference gateway for the first time.
